src/renderer/wg_engine/tvgWgGeometry.h

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright (c) 2023 the ThorVG project. All rights reserved.
+
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+#ifndef _TVG_WG_GEOMETRY_H_
+#define _TVG_WG_GEOMETRY_H_
+
+#include <cassert>
+#include <cmath>
+#include "tvgCommon.h"
+#include "tvgArray.h"
+
+class WgPoint
```

Comment:
tvg::Point / WgPoint / GlPoint... I believe it's time to refactor our fundamental structures.

Aside from this PR,  I think we consolidate general 2D Point operators in common/tvgPoint.h and then use them across the GL and WG engines. @RuiwenTang @SergeyLebedkin

Reply:
Fully support this idia.
Also it will be helpful to create Mtexri3x3 and Matrix4x4, and Vector3 and Vector4 (or Point3 and Point4)

Reply:
Well, except for GLPoint and Matrix, I think most of the GPU vertex data generation, texture and buffer management code can be shared. The specific GPU backend only needs to upload this data and call drawElements.